packages/openapi-typescript-helpers/index.d.ts

```diff
@@ -7,6 +7,22 @@ export type OkStatus = 200 | 201 | 202 | 203 | 204 | 206 | 207 | "2XX";
 // biome-ignore format: keep on one line
 export type ErrorStatus = 500 | 501 | 502 | 503 | 504 | 505 | 506 | 507 | 508 | 510 | 511 | '5XX' | 400 | 401 | 402 | 403 | 404 | 405 | 406 | 407 | 408 | 409 | 410 | 411 | 412 | 413 | 414 | 415 | 416 | 417 | 418 | 420 | 421 | 422 | 423 | 424 | 425 | 426 | 427 | 428 | 429 | 430 | 431 | 444 | 450 | 451 | 497 | 498 | 499 | '4XX' | "default";
 
+/**
+ * 'default' returns every not explicitly defined status code
+ * '2XX' returns all 2XX status codes
+ * '4XX' returns all 4XX status codes
+ * '5XX' returns all 5XX status codes
+ */
```

Comment:
Note: IIUC this implementation has the limitation that status codes unknown to `openapi-fetch` have to be explicitly defined in the OpenAPI spec so they can be typed on the client side.

For example, `"2XX"` in the OpenAPI spec will not match `209`:

```
if (status === 209) {
  // `data` will be never
}
```

IMO this is a reasonable limitation / design decision.

Relevant excerpt from [RFC 7231](https://datatracker.ietf.org/doc/html/rfc7231#section-6)

>    HTTP status codes are extensible.  HTTP clients are not required to
   understand the meaning of all registered status codes, though such
   understanding is obviously desirable.  However, a client MUST
   understand the class of any status code, as indicated by the first
   digit, and treat an unrecognized status code as being equivalent to
   the x00 status code of that class, with the exception that a
   recipient MUST NOT cache a response with an unrecognized status code.

>   For example, if an unrecognized status code of 471 is received by a
   client, the client can assume that there was something wrong with its
   request and treat the response as if it had received a 400 (Bad
   Request) status code.  The response message will usually contain a
   representation that explains the status.